src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {

```

Comment:
No, `parseDeclarationFlags` might return `None` in which case we don't want to parse as a declaration (but rather as an expression because someone used a contextual keyword as an identifier).